pkgs/development/python-modules/wasmer/default.nix

```diff
@@ -0,0 +1,65 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, pipInstallHook
+, pythonImportsCheckHook
+, maturin
+, python3
+, isPy27
+, component ? "api"
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "wasmer";
+  version = "1.0.0-beta1";
+  disabled = isPy27;
+
+  src = fetchFromGitHub {
+    owner = "wasmerio";
+    repo = "wasmer-python";
+    rev = version;
+    sha256 = "0302lcfjlw7nz18nf86z6swhhpp1qnpwcsm2fj4avl22rsv0h78j";
+  };
+
+  cargoSha256 = "1m48x3klwdh6n49hmlx1s0b8z49srhr3iq8plbpy74dspgs6dbj6";
+
+  nativeBuildInputs = [ pipInstallHook maturin pythonImportsCheckHook ];
+
+  preBuild = ''
+    cd packages/${component}
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    maturin build --release --manylinux off --strip
+
+    runHook postBuild
+  '';
+
+  postBuild = ''
+    cd ../..
+  '';
+
+  installPhase = ''
```

Comment:
Is this phase needed? The hook should run by itself. it just needs the wheel moved in a prePhase.

Reply:
Also, I prefer the wheel to be build in a separate derivation, and then passed onto `buildPythonPackage` with format = "wheel";`. Ideally the rust builder would also consist of hooks that we could use instead. An alternative solution would be to apply `toPythonModule` to `wasmer`, but I'm of the opinion we should only use that for non-wheel builds.

Reply:
Should the maturin hook be standalone or on top of buildRustPackage or be part of buildRustPackage?

Reply:
Ideally build functions are composed of hooks, so the latter.

Reply:
I am going to pass the wheel through a let in here and working in another PR on the hook otherwise this one gets never merged.